<a href="https://colab.research.google.com/github/unknown-jun/Basic_of_Python/blob/master/16_Dec_2020.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#복습
- 빅데이터 수집을 위해 반드시 알아야할 필수 기술 ?  웹스크롤링

- 웹스크롤링을 위해서 사용한 파이썬 모듈 ?  beautiful soup 모듈

- 웹스크롤링을 편하게 잘 하기 위한 비법 ?
   크롬의 개발자 모드(F12) 로 들어가서 화살표를 이용해서 웹상에서 스크롤링하기 원하는 지점의
   html 클래스 이름을 빠르게 찾을 수 있습니다.

#142. 웹스크롤링 실전 1단계 (EBS 레이디 버그 게시판)
- 회사의 신제품이 출시 되었을 때 그 제품에 대한 사람들의 반응을 데이터 분석을 하고자 할때 웹스크롤링과 시각화를 이용 가능

In [ ]:
# 예제1. ebs 레이디버그 시청자 게시판의 url을 가지고 직접 html 문서를 내려 받을 수 있도록 코드를 구현
# 어제는 우리가 crl+s를 누르고 웹페이지를 저장하고 구현했지만 웹에서 바로 html문서를 내려받는 방법

from bs4 import BeautifulSoup
import urllib.request        # 웹상의 url을 파이썬이 인식할 수 있도록 해주는 모듈

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"

url = urllib.request.Request(list_url)    # 사람이 알아볼 수 있는 위의 url을 파이썬이 알아볼수 있도록 변환하는 첫번째 작업
result = urllib.request.urlopen(url).read().decode("utf-8")   # 위의 url의 html문서들을 result 변수에 담음

print(result)

In [ ]:
# 예제2 위에서 불러온 html 코드를 BeautifulSoup의 함수를 이용해서 웹스크롤링 할 수 있도록 BeautifulSoup에 파싱하시오

from bs4 import BeautifulSoup
import urllib.request        

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"

url = urllib.request.Request(list_url)    
result = urllib.request.urlopen(url).read().decode("utf-8")   

soup = BeautifulSoup( result, "html.parser")

print(soup)

In [ ]:
# 예제3. 지금 페이지의 시청자 게시판의 글 내용에 해당하는 부분의 태그와 클래스 이름을 알아내시오
# =>   < p class_="con"> 재미있다 </p>

In [6]:
# 예제4. p 태그 중에 class가 con에 해당하는 부분을 스크롤링하시오

from bs4 import BeautifulSoup
import urllib.request        

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"

url = urllib.request.Request(list_url)    
result = urllib.request.urlopen(url).read().decode("utf-8")   

soup = BeautifulSoup( result, "html.parser")
result2 = soup.find_all('p', class_='con')   # find만 썼으면 맨 처음 하나만 가지고 오는데 find_all은 p 태그에 class 이름 con에 해당하는 부분 모두를 가져옴

print(result2)

[<p class="con">
								
							    
							  	    재미있 다
							    
						    
						    </p>, <p class="con">
								
							    
							  	    어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??
							    
						    
						    </p>, <p class="con">
								
							    
							  	    〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!



〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 

방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!

2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!

고맙습니다~
							    
						    
						    </p>, <p class="con">
								
							    
							  	    아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다
							    
						    
						    </p>, <p class="con">
								
							    
							  	    2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~

							    
						    
						    </p>, <p class="con">
								
							    
							  	    레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!

							    
						    
						    </p>, <p class="con">
								
							    
							  	    이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?
			

In [8]:
# 예제 5. 위의 결과에서 html 문서 말고 한글 텍스트만 가져오시오 (위의 result2는 리스트입니다)

from bs4 import BeautifulSoup
import urllib.request        

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"

url = urllib.request.Request(list_url)    
result = urllib.request.urlopen(url).read().decode("utf-8")   

soup = BeautifulSoup( result, "html.parser")
result2 = soup.find_all('p', class_='con')  

for i in result2:
    print( i.get_text() )  # 텍스트만 가져오는 방법


								
							    
							  	    재미있 다
							    
						    
						    

								
							    
							  	    어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??
							    
						    
						    

								
							    
							  	    〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!



〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 

방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!

2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!

고맙습니다~
							    
						    
						    

								
							    
							  	    아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다
							    
						    
						    

								
							    
							  	    2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~

							    
						    
						    

								
							    
							  	    레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!

							    
						    
						    

								
							    
							  	    이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?
							    
						    
						    

								
							    
							  	    너모 재밌어요!!! 저 시즌 1부터 다 본방으로 챙겨본 미라큘러 입니다ㅋㅋㅋ 진짜  재밌구요 팬카페랑 이런데도 다 가입해있어

In [11]:
# 예제 6. 위에서 출력되고 있는 텍스트들이 좀 더 깔끔하게 나오게 하시오
from bs4 import BeautifulSoup
import urllib.request        

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"

url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")   

soup = BeautifulSoup( result, "html.parser")
result2 = soup.find_all('p', class_='con')  

for i in result2:
    print( i.get_text(" ", strip = True) )  # 공백을 기준으로 잘라서 텍스트를 정리하여 보여 줌

재미있 다
어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??
〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!



〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 

방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!

2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!

고맙습니다~
아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다
2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~
레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!
이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?
너모 재밌어요!!! 저 시즌 1부터 다 본방으로 챙겨본 미라큘러 입니다ㅋㅋㅋ 진짜  재밌구요 팬카페랑 이런데도 다 가입해있어요... 방도 레이디버그 사진으로 덕-질☆
완전 재밌다
근뎅.....오즈 긑나서....레이디버그가 하는거잖아여....ㅠ
우왕!
넘재밌어요.매일하면 좋겠어요
재밌어요 완전 꿀잼이요 사랑합니다
ㅎㅎ
진짜여???
보니하니,레이디버그,형사 가제트,오즈,삐삐에 있었어요!
오늘부터 미라큘러스 시즌3 다시 방영한다네요!^^
목,금 저녁 7시
전 세매큐,픽시,가제트형사,오:마법을 찾아서,히어로써클,로빈후드에 있어여!ㅎㅎ즈
벌써 가을 인길요~
가을이 빨리 되면 좋겠어요


In [26]:
# 예제7. 위에서 출력되고 있는 텍스트들이 params 라는 비어있는 리스트에 담기게 하시오
from bs4 import BeautifulSoup
import urllib.request        

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"

url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")   

soup = BeautifulSoup( result, "html.parser")
result2 = soup.find_all('p', class_='con')  

params = []
for i in result2:
    params.append( i.get_text(" ", strip = True) )

print(params) 

['재미있 다', '어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??', '〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!\r\n\r\n\r\n\r\n〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 \r\n\r\n방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!\r\n\r\n2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!\r\n\r\n고맙습니다~', '아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다', '2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~', '레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!', '이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?', '너모 재밌어요!!! 저 시즌 1부터 다 본방으로 챙겨본 미라큘러 입니다ㅋㅋㅋ 진짜  재밌구요 팬카페랑 이런데도 다 가입해있어요... 방도 레이디버그 사진으로 덕-질☆', '완전 재밌다', '근뎅.....오즈 긑나서....레이디버그가 하는거잖아여....ㅠ', '우왕!', '넘재밌어요.매일하면 좋겠어요', '재밌어요 완전 꿀잼이요 사랑합니다', 'ㅎㅎ', '진짜여???', '보니하니,레이디버그,형사 가제트,오즈,삐삐에 있었어요!', '오늘부터 미라큘러스 시즌3 다시 방영한다네요!^^\r\n목,금 저녁 7시', '전 세매큐,픽시,가제트형사,오:마법을 찾아서,히어로써클,로빈후드에 있어여!ㅎㅎ즈', '벌써 가을 인길요~', '가을이 빨리 되면 좋겠어요']


In [27]:
# 예제8. 게시글을 올린 날짜를 스크롤링하기 위해 게시글 날짜가 있는 html문서의 태그이름과 클래스 이름을 확인하시오
#  => <span class="date">

In [28]:
# 예제9. 위의 날짜를 모두 스크롤링해서 param2라는 리스트에 담으시오

# 1. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request        

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"

url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")   
soup = BeautifulSoup( result, "html.parser")

# 2. 시청자 게시판의 날짜와 본문을 내용을 가져옴
result1 = soup.find_all('p', class_='con')
result2 = soup.find_all('span', class_='date')  

# 3. 시청자 게시판의 날짜와 본문을 params와 params 리스트에 담음
params1 = []
params2 = []

for i in result1:
    params1.append( i.get_text(" ", strip = True) )
for e in result2:
    params2.append( e.get_text(" ", strip = True) )
    
print(params1)
print(params2) 

['재미있 다', '어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??', '〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!\r\n\r\n\r\n\r\n〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 \r\n\r\n방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!\r\n\r\n2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!\r\n\r\n고맙습니다~', '아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다', '2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~', '레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!', '이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?', '너모 재밌어요!!! 저 시즌 1부터 다 본방으로 챙겨본 미라큘러 입니다ㅋㅋㅋ 진짜  재밌구요 팬카페랑 이런데도 다 가입해있어요... 방도 레이디버그 사진으로 덕-질☆', '완전 재밌다', '근뎅.....오즈 긑나서....레이디버그가 하는거잖아여....ㅠ', '우왕!', '넘재밌어요.매일하면 좋겠어요', '재밌어요 완전 꿀잼이요 사랑합니다', 'ㅎㅎ', '진짜여???', '보니하니,레이디버그,형사 가제트,오즈,삐삐에 있었어요!', '오늘부터 미라큘러스 시즌3 다시 방영한다네요!^^\r\n목,금 저녁 7시', '전 세매큐,픽시,가제트형사,오:마법을 찾아서,히어로써클,로빈후드에 있어여!ㅎㅎ즈', '벌써 가을 인길요~', '가을이 빨리 되면 좋겠어요']
['2020.12.11 19:52', '2020.12.11 19:06', '2020.12.11 01:35', '2020.12.10 12:43', '2020.10.17 15:30', '2020.10.16 12:25', '2020.10.15 21:04', '2020.09.29 14:46', '2020.09.29 11:30

In [33]:
# 예제 10. 위의 날짜와 본문 내용이 같이 출력되게 하시오

# 1. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request        

list_url = "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?hmpMnuId=106"

url = urllib.request.Request(list_url)
result = urllib.request.urlopen(url).read().decode("utf-8")   
soup = BeautifulSoup( result, "html.parser")

# 2. 시청자 게시판의 날짜와 본문을 내용을 가져옴
result1 = soup.find_all('p', class_='con')
result2 = soup.find_all('span', class_='date')  

# 3. 시청자 게시판의 날짜와 본문을 params와 params 리스트에 담음
params1 = []
params2 = []

for i in result1:
    params1.append( i.get_text(" ", strip = True) )
for e in result2:
    params2.append( e.get_text(" ", strip = True) )

# 4. 날짜와 본문을 같이 출력함.
for k,h in zip(params1, params2):
    print(k + '  ' +h)

재미있 다  2020.12.11 19:52
어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??  2020.12.11 19:06
〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!



〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 

방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!

2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!

고맙습니다~  2020.12.11 01:35
아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다  2020.12.10 12:43
2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~  2020.10.17 15:30
레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!  2020.10.16 12:25
이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?  2020.10.15 21:04
너모 재밌어요!!! 저 시즌 1부터 다 본방으로 챙겨본 미라큘러 입니다ㅋㅋㅋ 진짜  재밌구요 팬카페랑 이런데도 다 가입해있어요... 방도 레이디버그 사진으로 덕-질☆  2020.09.29 14:46
완전 재밌다  2020.09.29 11:30
근뎅.....오즈 긑나서....레이디버그가 하는거잖아여....ㅠ  2020.09.21 08:17
우왕!  2020.09.21 08:16
넘재밌어요.매일하면 좋겠어요  2020.09.20 09:07
재밌어요 완전 꿀잼이요 사랑합니다  2020.09.19 22:52
ㅎㅎ  2020.09.16 12:34
진짜여???  2020.09.14 10:26
보니하니,레이디버그,형사 가제트,오즈,삐삐에 있었어요!  2020.09.11 09:45
오늘부터 미라큘러스 시즌3 다시 방영한다네요!^^
목,금 저녁 7시  2020.09.11 09:42
전 세매큐,픽시,가제트형사,오:마법을 찾아서,히어로써클,로빈후드에 

**예제11. 아래의 결과를 출력하는데 페이지 번호가 1번부터 22번까지 변경되어서 출력되게 하시오!**

~~~
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.***page=1***&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.***page=2***&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.***page=3***&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
~~~

In [36]:
for i in range(1, 23):
    print( "http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page="+str(i)+"&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&")

http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=1&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=2&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=3&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=4&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=5&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&
http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=6&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeywor

**(중요!) 예제 12. 예제11번 코드를 예제 10번코드에 적용해서 레이디 버그 전체 게시판의 글들이 날짜와 함께 출력되게하시오**

In [40]:
# 1. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
from bs4 import BeautifulSoup
import urllib.request        


for i in range(1, 23):
    list_url="http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page="+str(i)+"&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&"

    url = urllib.request.Request(list_url)
    result = urllib.request.urlopen(url).read().decode("utf-8")   
    soup = BeautifulSoup( result, "html.parser")

    # 2. 시청자 게시판의 날짜와 본문을 내용을 가져옴
    result1 = soup.find_all('p', class_='con')
    result2 = soup.find_all('span', class_='date')  

    # 3. 시청자 게시판의 날짜와 본문을 params와 params 리스트에 담음
    params1 = []
    params2 = []

    for i in result1:
        params1.append( i.get_text(" ", strip = True) )
    for e in result2:
        params2.append( e.get_text(" ", strip = True) )

    # 4. 날짜와 본문을 같이 출력함.
    for k,h in zip(params1, params2):
        print(k + '  ' +h)

재미있 다  2020.12.11 19:52
어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??  2020.12.11 19:06
〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!



〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 

방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!

2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!

고맙습니다~  2020.12.11 01:35
아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다  2020.12.10 12:43
2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~  2020.10.17 15:30
레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!  2020.10.16 12:25
이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?  2020.10.15 21:04
너모 재밌어요!!! 저 시즌 1부터 다 본방으로 챙겨본 미라큘러 입니다ㅋㅋㅋ 진짜  재밌구요 팬카페랑 이런데도 다 가입해있어요... 방도 레이디버그 사진으로 덕-질☆  2020.09.29 14:46
완전 재밌다  2020.09.29 11:30
근뎅.....오즈 긑나서....레이디버그가 하는거잖아여....ㅠ  2020.09.21 08:17
우왕!  2020.09.21 08:16
넘재밌어요.매일하면 좋겠어요  2020.09.20 09:07
재밌어요 완전 꿀잼이요 사랑합니다  2020.09.19 22:52
ㅎㅎ  2020.09.16 12:34
진짜여???  2020.09.14 10:26
보니하니,레이디버그,형사 가제트,오즈,삐삐에 있었어요!  2020.09.11 09:45
오늘부터 미라큘러스 시즌3 다시 방영한다네요!^^
목,금 저녁 7시  2020.09.11 09:42
전 세매큐,픽시,가제트형사,오:마법을 찾아서,히어로써클,로빈후드에 

- 위의 params1과 params2 리스트가 for 문 안쪽에 있기 때문에 페이지 번호가 돌 때 마다 params 리스트 안의 내용이 새로운 날짜와 내용으로 변경되게 됨

예제 14. params1과 params2에 레이디 버그 시청자 게시판의 모든 게시날짜와 본문 내용이 들어가게 코드를 수정하시오

In [41]:
# 0. 웹스크롤링에 필요한 모듈을 import 함
from bs4 import BeautifulSoup
import urllib.request        

# 1. 레이디 버그 게시판 게시날짜와 게시글 전체를 다 담은 리스트 2개를 담음
params1 = [] # 게시 본문를 담음
params2 = [] # 게시 날짜를 담음


for i in range(1, 23):  # 페이지 번호를 1번 부터 22번까지 변경하기 위해 for 문을 사용
    
    # 2. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
    list_url="http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page="+str(i)+"&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&"

    url = urllib.request.Request(list_url)

    result = urllib.request.urlopen(url).read().decode("utf-8")   
    soup = BeautifulSoup( result, "html.parser")

    # 3. 시청자 게시판의 날짜와 본문을 내용을 가져옴
    result1 = soup.find_all('p', class_='con')
    result2 = soup.find_all('span', class_='date')  

    # 4. 시청자 게시판의 날짜와 본문을 params와 params 리스트에 담음
    # 그전 절차에선 이 자리에 비어있는 리스트가 있어 페이지번호가 바뀔때 마다 params리스트가 초기화가 되었음
    # 지금은 맨 위(for문 시작하기 전)에 있으므로 계속 페이지 1부터 22번까지의 모든 게시날짜와 게시본문이 리스트에 append 됨
    for i in result1:
        params1.append( i.get_text(" ", strip = True) )
    for e in result2:
        params2.append( e.get_text(" ", strip = True) )

# 5. 날짜와 본문을 같이 출력함.
for k,h in zip(params1, params2):
    print(k + '  ' +h)


재미있 다  2020.12.11 19:52
어제 흐름상 종영 같지 않았는데 갑자기 끝이네요??  2020.12.11 19:06
〈미라큘러스: 레이디버그와 블랙캣〉종영 안내!



〈미라큘러스: 레이디버그와 블랙캣〉-시즌3가 

방송 종료되었습니다, 그동안 시청해주셔서 감사드립니다!

2021년 새로운 시즌으로 여러분을 찾아뵙도록 노력하겠습니다!

고맙습니다~  2020.12.11 01:35
아이들 때문에 보게 되었는데 어른인 제가 더 좋아하는 프로그램이 되었네요 아이들이랑 볼 때 너무 신나요 함께 볼 수 있는 보기 드문 애니매이션이라 더 좋네요 시즌 4도 너무 너무 기대되고 기다리고 있습니다  2020.12.10 12:43
2021년이 빨리되서 레이디버그 시즌4를 보면 좋겠네요~  2020.10.17 15:30
레이디버그... 제가 어렸을때도 너~~무 좋아 하던 방송 이에요!!!!!!!  2020.10.16 12:25
이렇게 재미있는프로는 이 세상에 하나뿐이예요 아 그리고 뉴욕편은 언제 올려주시나요?  2020.10.15 21:04
너모 재밌어요!!! 저 시즌 1부터 다 본방으로 챙겨본 미라큘러 입니다ㅋㅋㅋ 진짜  재밌구요 팬카페랑 이런데도 다 가입해있어요... 방도 레이디버그 사진으로 덕-질☆  2020.09.29 14:46
완전 재밌다  2020.09.29 11:30
근뎅.....오즈 긑나서....레이디버그가 하는거잖아여....ㅠ  2020.09.21 08:17
우왕!  2020.09.21 08:16
넘재밌어요.매일하면 좋겠어요  2020.09.20 09:07
재밌어요 완전 꿀잼이요 사랑합니다  2020.09.19 22:52
ㅎㅎ  2020.09.16 12:34
진짜여???  2020.09.14 10:26
보니하니,레이디버그,형사 가제트,오즈,삐삐에 있었어요!  2020.09.11 09:45
오늘부터 미라큘러스 시즌3 다시 방영한다네요!^^
목,금 저녁 7시  2020.09.11 09:42
전 세매큐,픽시,가제트형사,오:마법을 찾아서,히어로써클,로빈후드에 

**문제 415. 레이디 버그 게시판의 전체 게시글은 총 몇건인가?**

In [43]:
# 0. 웹스크롤링에 필요한 모듈을 import 함
from bs4 import BeautifulSoup
import urllib.request        

# 1. 레이디 버그 게시판 게시날짜와 게시글 전체를 다 담은 리스트 2개를 담음
params1 = [] # 게시 본문를 담음
params2 = [] # 게시 날짜를 담음


for i in range(1, 23):  # 페이지 번호를 1번 부터 22번까지 변경하기 위해 for 문을 사용
    
    # 2. 웹에서 html 문서를 가져와 beautifulsoup으로 파싱
    list_url="http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page="+str(i)+"&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&"

    url = urllib.request.Request(list_url)

    result = urllib.request.urlopen(url).read().decode("utf-8")   
    soup = BeautifulSoup( result, "html.parser")

    # 3. 시청자 게시판의 날짜와 본문을 내용을 가져옴
    result1 = soup.find_all('p', class_='con')
    result2 = soup.find_all('span', class_='date')  

    # 4. 시청자 게시판의 날짜와 본문을 params와 params 리스트에 담음
    # 그전 절차에선 이 자리에 비어있는 리스트가 있어 페이지번호가 바뀔때 마다 params리스트가 초기화가 되었음
    # 지금은 맨 위(for문 시작하기 전)에 있으므로 계속 페이지 1부터 22번까지의 모든 게시날짜와 게시본문이 리스트에 append 됨
    for i in result1:
        params1.append( i.get_text(" ", strip = True) )
    for e in result2:
        params2.append( e.get_text(" ", strip = True) )

# 5. 본문 리스트의 갯수(구성요소들)를 파악
print( len(params1) )

429


In [ ]:
# 텍스트 파일을 저장하는 파이썬 코드

text ='abcd'

f= open('mytext32.txt','w',encoding='UTF8')
f.write( text )
f.close

**문제 416. 게시글 429개 전체를 c 드라이브 밑에 mytest34.txt라는 이름으로 저장하시오**

In [ ]:
# 1. 웹에서 html 문서를 가져와 beautifulsoup 으로 파싱 
from  bs4  import  BeautifulSoup
import  urllib.request   # 웹상의 url 을 파이썬이 인식할 수 있도록 해주는 모듈

params1 = []
params2 = []

f = open("/Users/Downloads/mytext34.txt")

for  i  in  range(1, 23):
    list_url = 'http://home.ebs.co.kr/ladybug/board/6/10059819/oneBoardList?c.page=' +str(i) + '&hmpMnuId=106&searchKeywordValue=0&bbsId=10059819&searchKeyword=&searchCondition=&searchConditionValue=0&'
    
    url = urllib.request.Request(list_url) #  사람이 알아볼 수 있는 위의 url 을 파이썬이 알아
                                                  # 볼 수 있도록 변환 첫번째 작업
    result = urllib.request.urlopen(url).read().decode("utf-8") 
    soup = BeautifulSoup( result, "html.parser")
    
    # 2. 시청자 게시판의 날짜와 본문 내용을 가져옵니다.
    result1 = soup.find_all( 'span', class_ ='date')
    result2 = soup.find_all( 'p',  class_ ='con') 
     
    # 3. 시청자 게시판의 날짜와 본문을 params 와 params2 리스트에 담습니다.
 
    for  i  in  result1:
        params1.append( i.get_text("  ", strip=True) )
    for  i  in  result2:
        params2.append( i.get_text("  ", strip=True) )
        
# 4. 날짜와 본문을 같이 저장 합니다.     
for   k,  h   in   zip(  params1,  params2 ):
    f.write ( k + '   '  + h + '\n' ) 

f.close()
       

#143. 웹스크롤링 실전2 ( 중앙일보사 )
- 중앙일보사 홈페이지에서 인공지능으로 검색을 했을 때 나오는 기사들을 전부 웹스크롤링 함.

예제1. 중앙일보사 홈페이지에서 인공지능으로 검색했을 때 나오는 url을 가져오시오!

=> https://news.joins.com/Search/JoongangNews?page=1&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews

예제2. 위의 사이트에서 보이는 상세기사를 클릭하고 그 기사의 url을 복사하시오

https://news.joins.com/article/23947044

https://news.joins.com/article/23946979

https://news.joins.com/article/23946876




예제3. 인공지능으로 검색했을 때 나오는 상세기사들의 url을 스크롤링하시오!

In [51]:
# 예제2 위에서 불러온 html 코드를 BeautifulSoup의 함수를 이용해서 웹스크롤링 할 수 있도록 BeautifulSoup에 파싱하시오

# 1. 웹스크롤링에 필요한 모듈을 import
from bs4 import BeautifulSoup
import urllib.request        

# 2. 중앙일보에서 인공지능으로 검색했을 나오는 첫페이지의 html코드를 BeautifulSoup 에서 이용할 수 있도록 파싱함
list_url = "https://news.joins.com/Search/JoongangNews?page=1&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews"

url = urllib.request.Request(list_url)    
result = urllib.request.urlopen(url).read().decode("utf-8")   

soup = BeautifulSoup( result, "html.parser")

# 3. 상세기사 url을 가져올 수 있도록 태그와 클래스를 찾음 (태그: h2, 클래스 이름: headline mg)
result1 = soup.find_all( 'h2', class_='headline mg')

# 4. 위의 result1은 리스트 이므로 for loop 문을 이용해 리스트에 있는 요소를 하나씩 빼내면서 href의 값을 얻어냄
for i in result1:   # result1 리스트 요소를 하나씩 빼내는 코드
    for k in i:     # h2 태그 안쪽 요소인 a태그의 html코드를 가져오기 위한 코드
        print(k.get('href'))    # a 태그의 href의 값을 얻어내라

https://news.joins.com/article/23947044
https://news.joins.com/article/23946979
https://news.joins.com/article/23946876
https://news.joins.com/article/23946841
https://news.joins.com/article/23946757
https://news.joins.com/article/23946642
https://news.joins.com/article/23946268
https://news.joins.com/article/23946216
https://news.joins.com/article/23946191
https://news.joins.com/article/23946032


예제 4. 위의 상세 기사 url을 params라는 비어있는 리스트에 다 append 시키시오

In [50]:
# 예제2 위에서 불러온 html 코드를 BeautifulSoup의 함수를 이용해서 웹스크롤링 할 수 있도록 BeautifulSoup에 파싱하시오

# 1. 웹스크롤링에 필요한 모듈을 import
from bs4 import BeautifulSoup
import urllib.request        

# 2. 중앙일보에서 인공지능으로 검색했을 나오는 첫페이지의 html코드를 BeautifulSoup 에서 이용할 수 있도록 파싱함
list_url = "https://news.joins.com/Search/JoongangNews?page=1&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews"

url = urllib.request.Request(list_url)    
result = urllib.request.urlopen(url).read().decode("utf-8")   

soup = BeautifulSoup( result, "html.parser")

# 3. 상세기사 url을 가져올 수 있도록 태그와 클래스를 찾음 (태그: h2, 클래스 이름: headline mg)
result1 = soup.find_all( 'h2', class_='headline mg')

# 4. 위의 result1은 리스트 이므로 for loop 문을 이용해 리스트에 있는 요소를 하나씩 빼내면서 href의 값을 얻어냄

params = []
for i in result1:   # result1 리스트 요소를 하나씩 빼내는 코드
    for k in i:     # h2 태그 안쪽 요소인 a태그의 html코드를 가져오기 위한 코드
        params.append(k.get('href'))    # a 태그의 href의 값을 얻어내라
print(params)

['https://news.joins.com/article/23947044', 'https://news.joins.com/article/23946979', 'https://news.joins.com/article/23946876', 'https://news.joins.com/article/23946841', 'https://news.joins.com/article/23946757', 'https://news.joins.com/article/23946642', 'https://news.joins.com/article/23946268', 'https://news.joins.com/article/23946216', 'https://news.joins.com/article/23946191', 'https://news.joins.com/article/23946032']


예제5. 상세기사 url 중에 하나를 복사해 오고 그 상세기사 url의 웹페이지로 접속해서 본문 기사의 태그 이름과 클래스 이름이 무엇인지 확인하시오!
~~~
태그 이름: div
클래스 이름: article_body mg fs4"
~~~

예제6. 위의 상세기사의 본문 텍스트를 스크롤링해서 출력하시오

In [54]:
# 1. 웹스크롤링에 필요한 모듈을 import
from bs4 import BeautifulSoup
import urllib.request        

# 2. 중앙일보에서 상세기사 url으로 검색했을 나오는 페이지의 html코드를 BeautifulSoup 에서 이용할 수 있도록 파싱함
list_url = "https://news.joins.com/article/23947044"

url = urllib.request.Request(list_url)    
result = urllib.request.urlopen(url).read().decode("utf-8")   

soup = BeautifulSoup( result, "html.parser")

# 3. 상세기사 본문을 가져올 수 있도록 태그와 클래스를 찾음 (태그: div, 클래스 이름: article_body mg fs4)
result1 = soup.find_all( 'div', class_='article_body mg fs4')

# 4. 위의 result1은 리스트 이므로 for loop 문을 이용해 리스트에 있는 요소를 하나씩 빼내면서 본문의 텍스트의 값을 얻어냄

params2 = []
for i in result1:   # result1 리스트 요소를 하나씩 빼내는 코드
    params2.append( i.get_text(" ", strip=True) )

print( params2 )

['김재철 동원그룹 명예회장이 16일 대전 KAIST 본원에서 열린 \'AI 발전기금 기부 약정식\'에서 500억원 기부를 발표했다. 프리랜서 김성태 “세계 역사는 대항해시대를 거치면서 패권의 큰 흐름이 바뀌었다. 이제 과거 모든 변화를 아우르는 21세기 4차 산업혁명의 시대에는 인공지능(AI)을 지배하는 자가 세계패권을 쥐게 된다.” “세계 역사는 대항해시대 이후 패권 바뀌어 21세기는 AI 지배하는 자가 세계패권 쥔다” KAIST 올 한해 총 1474억원 모아, 역대 최대 ‘마도로스’김재철(84) 동원그룹 명예회장이 AI 인재육성과 연구개발을 위해 \xa0KAIST에 500억원을 기부했다. \xa0KAIST는 16일 대전 본원에서 동원그룹 김재철 명예회장의 ‘AI 발전기금 기부 약정식’을 했다고 밝혔다. 약정식에는 신성철 총장 등 KAIST 주요 보직 교수들과 김 명예회장, 김 회장의 두 아들 김남구 한국금융지주 회장, 김남정 동원그룹 부회장 등이 참석했다. 김 명예회장은 약정식에서 “AI 물결이 대항해시대와 1ㆍ2ㆍ3차 산업혁명 이상으로 우리의 삶을 바꾸는 큰 변화를 이끌 것”이라며 “오늘 이 자리는 대한민국이 AI 선진국으로 나아가는 출정식이 될 것”이라고 말했다. 김 명예회장은 또 “위대한 잠재력을 가진 우리 국민이 국력을 모아 경쟁에 나서면 AI 선진국이 될 수 있다”며 “과학영재와 우수한 교수진이 집결해 있는 KAIST가 선두주자로서 우리나라 AI 개발 속도를 촉진하는 플래그십 역할을 할 것”이라고 덧붙였다. KAIST AI대학원은 지난해 3월 국내 대학 중 최초로 과학기술정보통신부의 AI대학원 지원사업에 선정됐으며, 5개월 뒤인 8월 문을 열었다. 지난달 말 현재 구글ㆍIBM왓슨ㆍ마이크로소프트 등 글로벌 정보기술(IT) 기업의 AI연구소 출신 전임교수 13명과 겸임교수 8명 등 21명으로 교수진을 구성하고 있다. 내년 3월부터는 단계적으로 캠퍼스를 대전 본원에서 서울 홍릉으로 이전할 계획이다. 또 2023년에는 서울 양재동 ‘양재 연구개발 혁신지

예제 7. 상세 기사 url을 가져와서 params 리스트에 넣고 출력하는 예제 4번 코드를 함수로 생성하시오

~~~
print( j_scroll() )
['https://news.joins.com/article/23947044', 'https://news.joins.com/article/23946979', ... ,'https://news.joins.com/article/23946191', 'https://news.joins.com/article/23946032']
~~~

In [58]:
from bs4 import BeautifulSoup
import urllib.request 

def j_scroll():      
    list_url = "https://news.joins.com/Search/JoongangNews?page=1&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews"

    url = urllib.request.Request(list_url)    
    result = urllib.request.urlopen(url).read().decode("utf-8")   

    soup = BeautifulSoup( result, "html.parser")

    result1 = soup.find_all( 'h2', class_='headline mg')

    params = []
    for i in result1:  
        for k in i:    
            params.append(k.get('href'))  

    return params
print( j_scroll() )

['https://news.joins.com/article/23947044', 'https://news.joins.com/article/23946979', 'https://news.joins.com/article/23946876', 'https://news.joins.com/article/23946841', 'https://news.joins.com/article/23946757', 'https://news.joins.com/article/23946642', 'https://news.joins.com/article/23946268', 'https://news.joins.com/article/23946216', 'https://news.joins.com/article/23946191', 'https://news.joins.com/article/23946032']


예제 8. 상세 기사 url로 본문 기사를 스크롤링해서 리스트에 담았던 예제 6번을 j_detail_scroll() 함수로 생성하시오

In [59]:
from bs4 import BeautifulSoup
import urllib.request        

def j_detail_scroll():

    list_url = "https://news.joins.com/article/23947044"

    url = urllib.request.Request(list_url)    
    result = urllib.request.urlopen(url).read().decode("utf-8")   

    soup = BeautifulSoup( result, "html.parser")

    result1 = soup.find_all( 'div', class_='article_body mg fs4')

    params2 = []
    for i in result1:   # result1 리스트 요소를 하나씩 빼내는 코드
        params2.append( i.get_text(" ", strip=True) )

    return ( params2 )

print( j_detail_scroll() )

['김재철 동원그룹 명예회장이 16일 대전 KAIST 본원에서 열린 \'AI 발전기금 기부 약정식\'에서 500억원 기부를 발표했다. 프리랜서 김성태 “세계 역사는 대항해시대를 거치면서 패권의 큰 흐름이 바뀌었다. 이제 과거 모든 변화를 아우르는 21세기 4차 산업혁명의 시대에는 인공지능(AI)을 지배하는 자가 세계패권을 쥐게 된다.” “세계 역사는 대항해시대 이후 패권 바뀌어 21세기는 AI 지배하는 자가 세계패권 쥔다” KAIST 올 한해 총 1474억원 모아, 역대 최대 ‘마도로스’김재철(84) 동원그룹 명예회장이 AI 인재육성과 연구개발을 위해 \xa0KAIST에 500억원을 기부했다. \xa0KAIST는 16일 대전 본원에서 동원그룹 김재철 명예회장의 ‘AI 발전기금 기부 약정식’을 했다고 밝혔다. 약정식에는 신성철 총장 등 KAIST 주요 보직 교수들과 김 명예회장, 김 회장의 두 아들 김남구 한국금융지주 회장, 김남정 동원그룹 부회장 등이 참석했다. 김 명예회장은 약정식에서 “AI 물결이 대항해시대와 1ㆍ2ㆍ3차 산업혁명 이상으로 우리의 삶을 바꾸는 큰 변화를 이끌 것”이라며 “오늘 이 자리는 대한민국이 AI 선진국으로 나아가는 출정식이 될 것”이라고 말했다. 김 명예회장은 또 “위대한 잠재력을 가진 우리 국민이 국력을 모아 경쟁에 나서면 AI 선진국이 될 수 있다”며 “과학영재와 우수한 교수진이 집결해 있는 KAIST가 선두주자로서 우리나라 AI 개발 속도를 촉진하는 플래그십 역할을 할 것”이라고 덧붙였다. KAIST AI대학원은 지난해 3월 국내 대학 중 최초로 과학기술정보통신부의 AI대학원 지원사업에 선정됐으며, 5개월 뒤인 8월 문을 열었다. 지난달 말 현재 구글ㆍIBM왓슨ㆍ마이크로소프트 등 글로벌 정보기술(IT) 기업의 AI연구소 출신 전임교수 13명과 겸임교수 8명 등 21명으로 교수진을 구성하고 있다. 내년 3월부터는 단계적으로 캠퍼스를 대전 본원에서 서울 홍릉으로 이전할 계획이다. 또 2023년에는 서울 양재동 ‘양재 연구개발 혁신지

예제9. 지금 만든 j_detail_scroll() 함수는 상세기사 딱 한개의 본문을 출력하는 함수인데  
이 j_detail_scroll() 함수에 j_scroll() 함수를 실행했을 때 나오는  
상세 url 여러개를 제공할 수 있도록 코드를 수정하시오

In [61]:
from bs4 import BeautifulSoup
import urllib.request

def j_scroll():
    """
    중앙일보 기사에서 인공지능을 검색했을 때 전체적으로 나오는 url을 params 에 담는 함수
    """
    list_url = "https://news.joins.com/Search/JoongangNews?page=1&Keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&SortType=New&SearchCategoryType=JoongangNews"

    url = urllib.request.Request(list_url)    
    result = urllib.request.urlopen(url).read().decode("utf-8")   

    soup = BeautifulSoup( result, "html.parser")

    result1 = soup.find_all( 'h2', class_='headline mg')

    params = []
    for i in result1:  
        for k in i:    
            params.append(k.get('href'))  

    return params

def j_detail_scroll():
    """
    중앙일보에서 상세기사로 들어갔을때
    그 url에서 본문 기사들을 추출하는 함수
    """
    list_url = j_scroll()
    params2 = []
    
    for i in list_url:
        url = urllib.request.Request(i)    
        result = urllib.request.urlopen(url).read().decode("utf-8")   

        soup = BeautifulSoup( result, "html.parser")

        result1 = soup.find_all( 'div', class_='article_body mg fs4')

        for i in result1:   
            params2.append( i.get_text(" ", strip=True) )

    return ( params2 )

print( j_detail_scroll() )

['명지대학교(총장 유병진) ‘명지스마트팩토리(MSF) 개소식’이 12월 16일(수) 오전 10시 30분 명지대학교 제1공학관 B1층 MSF(Y024호)에서 개최됐다. 명지스마트팩토리는 4차산업혁명의 미래형 산업변화에 대응하기 위해서 스마트 임베디드 플랫폼 사업단(SEP)에서 2019년 3월부터 교육부 대학혁신지원사업의 일환으로 기획 및 구축해왔다. 이번 행사는 코로나19의 확산을 방지하기 위해 유병진 총장을 비롯하여 소수의 관계자만 참석하였으며, 정부의 방역지침 및 사회적 거리두기 지침을 준수하며 온라인 커뮤니케이션 플랫폼 줌(Zoom)으로 진행되었다. ‘명지스마트팩토리(MSF) 개소식’은 개회, 기도(김진옥 교목), 참석자 소개, 축사(유병진 총장), MSF 구축 성과 보고, 테이프 커팅. MSF 시연 및 질의응답, 맺음말, \xa0폐회, 단체 기념촬영 순으로 진행되었다. 명지스마트팩토리(MSF) 구축 사업에는 3년간 약 11억 원의 재정지원 사업비가 투입되었으며, 스마트팩토리 설계와 시설공사 및 기자재 구매를 진행하여 4차산업혁명 시대에 요구되는 첨단기술이 적용된 교육용 스마트팩토리를 구축하였다. 명지 스마트 팩토리에서는 지능형 임베디드 기전공학(스마트로봇, 스마트자동차, 인공지능 소프트웨어) 관련 실무교육을 강화하고 4차산업 혁명시대를 선도하는 명지대학교의 브랜드로서의 역할을 강화한다. 유병진 총장은 축사에서 “오늘 개소하는 명지스마트팩토리는 4차산업혁명의 핵심인 스마트자동차와 스마트로봇으로 구성된 교육용 스마트팩토리로, 4차산업혁명의 신기술에 대한 요소 기술과 시스템 통합기술을 학습할 수 있는 시설이다”며 “명지스마트팩토리는 향후 4차산업혁명 시대를 선도하는 유능한 융합인재를 육성할 수 있는 훌륭한 교육의 장이 될 것이다. 첨단 기술들이 어떻게 작동하고, 어떻게 연결되는지 몸소 체험하며 훌륭한 인재로 발돋움하게 될 것”이라고 밝혔다. 스마트 임베디드 플랫폼 사업단(SEP) 박강 단장은 “현재 4차산업혁명이 전 세계적으로 일어나고, 더 빠른 속도로 우리

# 중앙일보의 기사를 스크롤링 하는 내용 총정리
1. 검색 키워드(예: 인공지능)를 가지고 검색을 한 후 그 url을 얻어낸다.
2. 상세기사 url을 리스트에 담는 j_scroll() 함수를 생성한다.
3. 상세기사 url로 기사 본문을 스크롤링하는 j_detail_scroll() 함수를 생성한다.
4. j_detail_scroll() 함수 안에 j_scroll() 함수를 호출해서 상세기사 url 여러개를 받아오게끔 코딩하고 받아온 상세기사 url로 본문 기사들을 params2에 담게끔 코드를 수정함.

**문제. 417. 동아일보에서 검색 키워드 '인공지능'을 가지고 검색을 한 후 그 url을 얻어내라**

~~~
https://www.donga.com/news/search?p=1&query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&check_news=1&more=1&sorting=1&search_date=1&v1=&v2=&range=1
https://www.donga.com/news/search?p=16&query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&check_news=1&more=1&sorting=1&search_date=1&v1=&v2=&range=1
https://www.donga.com/news/search?p=31&query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&check_news=1&more=1&sorting=1&search_date=1&v1=&v2=&range=1
~~~

**문제. 418 상세기사 url을 리스트에 담는 d_scroll() 함수를 생성하세요**

In [66]:
from bs4 import BeautifulSoup
import urllib.request

def d_scroll():

    list_url = "https://www.donga.com/news/search?p=1&query=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&check_news=1&more=1&sorting=1&search_date=1&v1=&v2=&range=1"

    url = urllib.request.Request(list_url)    
    result = urllib.request.urlopen(url).read().decode("utf-8")   

    soup = BeautifulSoup( result, "html.parser")

    result1 = soup.find_all( 'p', class_='txt')

    params = []
    for i in result1:  
        for k in i:    
            params.append(k.get('href'))  

    return params

d_scroll()


['https://www.donga.com/news/article/all/20201216/104475349/1',
 'https://www.donga.com/news/article/all/20201216/104472426/2',
 'https://www.donga.com/news/article/all/20201216/104471612/1',
 'https://www.donga.com/news/article/all/20201216/104471110/1',
 'https://www.donga.com/news/article/all/20201215/104463480/1',
 'https://www.donga.com/news/article/all/20201215/104463391/1',
 'https://www.donga.com/news/article/all/20201215/104449354/1',
 'https://www.donga.com/news/article/all/20201215/104461113/1',
 'https://www.donga.com/news/article/all/20201215/104459002/1',
 'https://www.donga.com/news/article/all/20201215/104454507/1',
 'https://www.donga.com/news/article/all/20201215/104448246/1',
 'https://www.donga.com/news/article/all/20201215/104448168/2',
 'https://www.donga.com/news/article/all/20201215/104447953/2',
 'https://www.donga.com/news/article/all/20201214/104445042/1',
 'https://www.donga.com/news/article/all/20201214/104444592/1']

**문제 419. 상세기사 url로 기사본문을 스크롤링하는 d_detail_scroll() 함수를 생성하시오**

In [67]:
def d_detail_scroll():
    """
    중앙일보에서 상세기사로 들어갔을때
    그 url에서 본문 기사들을 추출하는 함수
    """
    list_url = d_scroll()
    params2 = []
    
    for i in list_url:
        url = urllib.request.Request(i)    
        result = urllib.request.urlopen(url).read().decode("utf-8")   

        soup = BeautifulSoup( result, "html.parser")

        result1 = soup.find_all( 'div', class_='article_txt')

        for i in result1:   
            params2.append( i.get_text(" ", strip=True) )

    return ( params2 )

print( d_detail_scroll() )

["“드디어 만선을 했다. 작열하는 태양의 밑에서 또는 폭풍우 속에서 위험을 무릅쓰고 한 마리 한 마리 가슴을 졸이며 낚아 올린 고기가 이젠 어창에 그득히 차서…어렵게 낚은 다랑어를 흉측한 돌고래나 모진 상어 떼에 빼앗긴 적도 많지만….” 1975년 판 실업계 고교 2학년 국어교과서(문교부 발행)에 실린 ‘거친 파도를 헤치고’란 글이다. 소설 ‘노인과 바다’가 떠오르는 이 글은 ‘참치’하면 생각나는 동원그룹의 김재철 명예회장이 썼다. 1966년 고려원양 광명호 선장 시절에 일기 형식으로 쓴 글로 문장이 빼어나 교과서에도 실렸다. 대양을 누비며 동원그룹을 일군 그의 열정은 이제 고기의 바다에서 ‘데이터의 바다’로 바뀌었다. 김 명예회장은 “젊은 시절엔 세계의 푸른 바다에서 대한민국의 미래를 찾았지만, 인공지능(AI) 시대에는 데이터의 바다에 새로운 미래가 있을 것”이라며 AI 인재 육성에 써달라는 당부와 함께 KAIST에 사재 500억원을 출연했다. window.googletag = window.googletag || {cmd: []};\r\n          googletag.cmd.push(function() {\r\n            googletag.defineSlot('/1249652/Donga_OSV_Desktop_1x1', [1, 1], 'div-gpt-ad-1595277065244-0').addService(googletag.pubads());\r\n            googletag.pubads().enableSingleRequest();\r\n            googletag.pubads().collapseEmptyDivs();\r\n            googletag.enableServices();\r\n          }); googletag.cmd.push(function() { googletag.display('div-gpt-ad-1595277065244-0'); }); 16일 오전 대전 KAIST 본원 학술문화관 정근

**문제 420. 한겨례 신문사에서 인공지능으로 검색했을 때 나오는 기사 본문을 스크롤링하는 함수 두개를 생성하시오!**
~~~
1. h_scroll() : 상세기사 url을 가져오는 함수
2. h_detail_scroll(): 상세기사 url로 기사본문을 가져오는 함수
~~~

In [76]:
from bs4 import BeautifulSoup
import urllib.request

def h_scroll():

    list_url = "http://search.hani.co.kr/Search?command=query&keyword=%EC%9D%B8%EA%B3%B5%EC%A7%80%EB%8A%A5&media=news&submedia=&sort=d&period=all&datefrom=1988.01.01&dateto=2020.12.16&pageseq=0"

    url = urllib.request.Request(list_url)    
    result = urllib.request.urlopen(url).read().decode("utf-8")   

    soup = BeautifulSoup( result, "html.parser")

    result1 = soup.find_all( 'dt' )

    params = []
    for i in result1:  
        for k in i:    
            params.append('http:' + k.get('href'))  

    return params


def h_detail_scroll():
    list_url = h_scroll()
    params2 = []
    
    for i in list_url:
        url = urllib.request.Request(i)    
        result = urllib.request.urlopen(url).read().decode("utf-8")   

        soup = BeautifulSoup( result, "html.parser")

        result1 = soup.find_all( 'div', class_='text')

        for i in result1:   
            params2.append( i.get_text(" ", strip=True) )

    return ( params2 )

print( h_detail_scroll() )


["◇ 육군은 15일 국립대전현충에서 ‘6·25 전사자 발굴 유해 합동 안장식'을 엄수했다. 안장된 전사자는 고 배석래 이등중사, 송해경 이등중사, 이형술 하사, 문장춘 일병, 명한협 일병, 김학제 일병이다. ◇ 국립광주박물관·임방울국악진흥회는 19일 ‘국창 임방울의 소리향연'을 한다. 김천농악단의 오형우 외 6명의 풍물 판굿 무대, 부채춤, 판소리, 퓨전국악 등을 선보인다. 오프라인 관람은 누리집(gwangju.museum.go.kr)에서 사전 예약을 하면 되고, 유튜브 채널(youtube.com/channel/UCyZbaQ52Gh8SBr6GDdYx-Bw)에서 생중계한다. ◇ 한국언론진흥재단은 16일 언론의 신뢰 회복과 저널리즘 구현을 위한 ‘언론윤리헌장(안)' 제정 공청회를 온라인으로 한다. 언론윤리헌장(안)의 제정 경위와 경과보고, 주요 내용 설명, 언론 3단체 입장 발표 등이 진행되며, 언론진흥재단 유튜브 채널(youtube.com/channel/UCK9lNxYzWToTDhozDv6MSlA)에서 볼 수 있다. ◇ 넥슨재단은 푸르메재단 넥슨어린이재활병원에 병원 내 감염관리 체계 강화를 위한 운영 기금 3억820만원을 기부했다. 기금에는 지난달 열린 넥슨 콘텐츠 축제 ‘네코제9'의 판매 수익금이 포함됐다. ◇ 과학기술정보통신부(장관 최기영)는 16일 ‘2020 ICT 기기 산업 페스티벌'을 온라인으로 한다. 인공지능 기술이 적용된 컴퓨팅 서버 등 국내에서 개발된 신기술과 제품을 선보이는 전시회와 기술 동향과 시장 발전 방향을 논의하는 컨퍼런스를 한다. 누리집(ictfestival.kr)에서 사전등록 후 무료로 볼 수 있다. ◇ 중소기업중앙회(회장 김기문)는 15일 소기업·소상공인을 위한 공적 공제 제도 ‘노란우산'의 홍보대사로 배우 김영철과 가수 송가인을 위촉했다. ◇ 행정안전부 대구청사관리소(소장 이상수)는 15일 인사혁신처가 주관한 ‘2020년 공무원 재능나눔·자원봉사 우수사례'에서 최우수기관으로 선정됐다. 2016년부터 관내 소외계층을 대상으로 진행해

**문제 421. 전자신문사에서 스크롤링하고 싶은 키워드를 넣고 검색했을 때 나오는 본문 기사들을 수집하는 함수 2개를 생성하시오**